lib/elixir/lib/module.ex

```diff
@@ -1246,16 +1256,17 @@ defmodule Module do
     end
   end
 
-  defp compile_doc(table, line, kind, pair, args, doc, env) do
+  defp compile_doc(table, line, kind, pair, args, doc, env, impl) do
     signature = build_signature(args, env)
 
     case :ets.lookup(table, {:doc, pair}) do
       [] ->
+        doc = if impl, do: false, else: doc
         :ets.insert(table, {{:doc, pair}, line, kind, signature, doc})
 
       [{doc_tuple, line, _current_kind, current_sign, current_doc}] ->
         signature = merge_signatures(current_sign, signature, 1)
-        doc = if is_nil(doc), do: current_doc, else: doc
+        doc = if is_nil(doc) || impl, do: current_doc, else: doc
```

Comment:
Same here. If there is a doc and impl then we keep the `doc`. I think we need to keep this line as is and then add the same line I suggested above to the line below:

```elixir
doc = if is_nil(doc) && impl, do: false, else: doc
```